sdk/src/verify.rs

```diff
@@ -47,64 +58,165 @@ pub fn setup_cpi_accounts<'info>(
             + InvokeCpiContextAccount<'info>
             + Bumps,
     >,
-) -> InvokeCpiInstruction<'info> {
-    InvokeCpiInstruction {
-        fee_payer: ctx.accounts.get_fee_payer().to_account_info(),
-        authority: ctx.accounts.get_authority().to_account_info(),
-        registered_program_pda: ctx.accounts.get_registered_program_pda().to_account_info(),
-        noop_program: ctx.accounts.get_noop_program().to_account_info(),
-        account_compression_authority: ctx
-            .accounts
+) -> (Vec<AccountInfo<'info>>, Vec<AccountMeta>) {
+    // The trick for having `None` accounts is to pass program ID, see
+    // https://github.com/coral-xyz/anchor/pull/2101
+    let none_account_info = ctx.accounts.get_light_system_program().to_account_info();
+
+    let (cpi_context_account_info, cpi_context_account_meta) =
+        match ctx.accounts.get_cpi_context_account() {
```

Comment:
There are more accounts that are optional, do we require these to be always `Some()` in the current implementation?

Reply:
ah we currently just set those accounts to `None` by default with `none_account_info`

Reply:
this is independent from token accounts this is for sol

Reply:
But it's still for compressed SOL transfers, right? And here, again, we need to design good abstractions.